replica/table.cc

```diff
@@ -97,7 +97,12 @@ lw_shared_ptr<sstables::sstable_set> compaction_group::make_compound_sstable_set
 }
 
 lw_shared_ptr<sstables::sstable_set> table::make_compound_sstable_set() {
-    return _compaction_group->make_compound_sstable_set();
+    if (auto cg = single_compaction_group_if_available()) {
+        return cg->make_compound_sstable_set();
+    }
+    auto sstable_sets = boost::copy_range<std::vector<lw_shared_ptr<sstables::sstable_set>>>(compaction_groups()
+        | boost::adaptors::transformed(std::mem_fn(&compaction_group::make_compound_sstable_set)));
+    return make_lw_shared(sstables::make_compound_sstable_set(schema(), std::move(sstable_sets)));
```

Comment:
We should consider a custom solution here, since this pins all of the compaction groups' sstables, whereas we could pin just one. But this is no worse than what we have now, so fine for now.

Reply:
This compound set spanning all groups is there for table operations which span all the groups. Point queries should pin a single sstable set. Range scans should still be able to incrementally read the sstables through the incremental selection procedure which compound set implements. Each compaction group implements its own incremental selector, and when reading from that compound set, the incremental selection still happens as expected as groups are disjoint in their token range.

Reply:
Look at compound set selector, it merges the results from all selectors rather than sorting them.

Reply:
It could use a heap to return them incrementally. But a tablet_sstable_set could just iterate based on the token.

Reply:
you're right. I assumed partitioned incremental selection, but I can't do that. Also, L0 sstables are included unconditionally. We can teach compound set selector to work with one selector at a time, based on current position.

Reply:
I'm guessing that a dedicated selector will be easier (but just a guess).